packages/backend/src/core/transaction-count/getFilledDailyCounts.ts

```diff
@@ -0,0 +1,20 @@
+import { UnixTime } from '@l2beat/types'
+
+import { fillMissingDailyCounts } from './fillMissingDailyCounts'
+import { DailyTransactionCount } from './TransactionCounter'
+
+export function getFilledDailyCounts(
+  counts: DailyTransactionCount[],
+  lastTimestamp: UnixTime,
+): DailyTransactionCount[] {
+  const tip = counts.at(-1)?.timestamp
+  const isFullySynced = tip && lastTimestamp.lte(tip)
+  const startOfDay = UnixTime.now().toStartOf('day')
```

Comment:
Shouldn't this rely on the clock?

Reply:
Passing it to the function indeed makes it easier to test. I'm not convinced we need clock here - could you provide an example of why lack of clock would introduce an error?